src/sqlfluff/core/config.py

```diff
@@ -685,21 +685,38 @@ def load_config_up_to_path(
             self.load_user_appdir_config() if not ignore_local_config else {}
         )
         user_config = self.load_user_config() if not ignore_local_config else {}
-        config_paths = (
-            iter_intermediate_paths(Path(path).absolute(), Path.cwd())
-            if not ignore_local_config
-            else {}
-        )
-        config_stack = (
-            [self.load_config_at_path(p) for p in config_paths]
-            if not ignore_local_config
-            else []
-        )
+        parent_config_stack = []
+        config_stack = []
+        if not ignore_local_config:
+            # Finding all paths between here and the home
+            # directory. We could start at the root of the filesystem,
+            # but depending on the user's setup, this might result in
+            # permissions errors.
+            parent_config_paths = list(
+                iter_intermediate_paths(Path(path).absolute(),
+                                        Path(os.path.expanduser("~")))
+            )
+            # Stripping off the home directory and the current working
+            # directory, since they are both covered by other code
+            # here
+            parent_config_paths = parent_config_paths[1:-1]
+            parent_config_stack = (
+                [self.load_config_at_path(p) for p in reversed(list(parent_config_paths))]
```

Comment:
This actually loads all configs it finds, but in the next step we only keep the first one. Maybe this can be more efficient by having an explicit for loop with a break statement? 🤔